pyDeltaRCM/preprocessor.py

```diff
@@ -93,6 +165,11 @@ def construct_job_list(self):
         self.job_list = []
         if self._has_matrix:
             self.expand_yaml_matrix()
+            # loop the expanded jobs
+            for i in range(len(self.file_list)):
+                self.job_list.append(self._Job(self.file_list[i],
```

Comment:
I'm not sure. I think as a user I would rather be told up front that the configuration is bad and that the suite of runs won't run as expected. Do you agree, or would you rather the runs that are able to run proceed? It's easy enough to implement the latter, but I think that could be confusing behavior...

I'll look at the specific error that is raised though, and maybe with a try-except for that specific error, we could provide a more helpful error message in the matrix expansion here.

Related: I have introduced a ton of `ValueError`s for the configuration...maybe we want to subclass the `Exception` with our own `YamlConfigurationError`. This would let us easily have a documentation page about how to handle each raised `YamlConfigurationError`. I'll probably open an issue about this, rather than address it now.

Reply:
> I think as a user I would rather be told up front that the configuration is bad and that the suite of runs won't run as expected.

Sure I can agree with this.